hera_sim/rfi.py

```diff
@@ -184,9 +189,65 @@ def rfi_scatter(fqs, lsts, rfi=None, chance=0.0001, strength=10, std=10):
     """
     if rfi is None:
         rfi = np.zeros((lsts.size, fqs.size), dtype=np.complex)
-    assert rfi.shape == (lsts.size, fqs.size)
+    assert rfi.shape == (lsts.size, fqs.size), "rfi shape is not (lsts.size, fqs.size)"
+
     rfis = np.where(np.random.uniform(size=rfi.size) <= chance)[0]
     rfi.flat[rfis] += np.random.normal(strength, std) * np.exp(
-        1j * np.random.uniform(size=rfis.size)
+        2 * np.pi * 1j * np.random.uniform(size=rfis.size)
     )
     return rfi
+
+
+def rfi_dtv(fqs, lsts, rfi=None, freq_min=.174, freq_max=.214, width=0.008,
+            chance=0.0001, strength=10, strength_std=10):
+    """
+    Generate an (NTIMES, NFREQS) waterfall containing Digital TV RFI.
+
+    DTV RFI is expected to be of uniform bandwidth (eg. 8MHz), in contiguous
+    bands, in a nominal frequency range. Furthermore, it is expected to be
+    short-duration, and so is implemented as randomly affecting discrete LSTS.
+
+    There may be evidence that close times are correlated in having DTV RFI,
+    and this is *not currently implemented*.
+
+    Args:
+        fqs (array-like): shape=(NFREQS,), GHz
+            the spectral frequencies of the waterfall to be generated.
+        lsts (array-like): shape=(NTIMES,), radians
+            local sidereal times of the waterfall to be generated.
+        rfi (array-like): shape=(NTIMES,NFREQS), default=None
+            an array to which the RFI will be added.  If None, a new array
+            is generated.
+        freq_min, freq_max (float):
+            the min and max frequencies of the full DTV band [GHz]
+        width (float):
+            Width of individual DTV bands [GHz]
+        chance (float): default=0.0001
+            the probability that a time/freq bin will be assigned an RFI impulse
+        strength (float): Jy, default=10
```

Comment:
Is this default (and strength_std below) in line with what @mwilensky768 sees in data?
In general we should evaluate some of these options at some point.

Reply:
Also, should we allow for an array input for the different channels?

Reply:
I actually thought about the strength/strength_std myself, but declined to comment in my review. I'll explain below.

As set here, the defaults are saying that the RFI-signal/RFI-noise ratio is 1. I certainly don't see that in SSINS, but the SSINS are a highly averaged-down space. Since I don't really work with the raw visibilities, I don't actually know what the RFI-signal/RFI-noise ratio is in the raw visibilities. Seeing as it gets caught by xrfi (single-baseline flagger) some of the time, I would hazard to guess that the ratio of RFI-signal/RFI-noise is greater than 1 in the raw visibilities.

Reply:
At least in the MWA, it's not uncommon for it to be undetectable in single visibilities (right @mwilensky768?)

Reply:
Hmm, couple thoughts here:
1) I'm not sure this is really a signal to noise ratio. It's a signal vs variation of the signal -- so sometimes it's bright and other times it's not. This is different than say thermal noise of the visibilities.
2) The xrfi we're running now (and where I've looked into this a little) is averaged across baselines (in a weird way), so not a single-baseline flagger. I also haven't looked back to the raw visibilities though.

Reply:
I guess that since sometimes it is detectable in the single visibilities and sometimes not, that a SNR of 1 is a good middle-ground default.

Reply:
Defaults were copied from ``rfi_scatter``, as I have no feel for what they should be. I think @adampbeardsley  is right that this is a good project.

I've updated code to allow array input for different bands in 62e00855410610f1326b5394411926f5a49e205a.

Reply:
@adampbeardsley if the different frequencies are drawn independently, then I'm not seeing the difference between variation on a signal and noise on a signal. Happy to table that for now though.

Reply:
I think what @adampbeardsley is getting at is that there is thermal noise in the visibility (potentially), and the std in this routine does not specify _that_ level of noise, just how much variation there is between levels of RFI "signal" for different LSTS/FREQS